ovs/extensions/healthcheck/generic_hc.py

```diff
@@ -439,3 +441,21 @@ def verify_rabbitmq(result_handler):
                 result_handler.failure('RabbitMQ has partition issues: {0}'.format(', '.join(partitions)))
         else:
             result_handler.skip('RabbitMQ is not running/active on this server!')
+
+    @staticmethod
+    @expose_to_cli(MODULE, 'recovery-domain-test', HealthCheckCLIRunner.ADDON_TYPE)
+    def check_recovery_domains(result_handler):
+        prim_domains = [i.name for i in DomainList.get_domains() if len(i.storage_router_layout['regular']) >= 1]
+        for domain in DomainList.get_domains():
+            layout = domain._storage_router_layout()
```

Comment:
Call it like a property (domain.storage_router_layout) to make use of the caching (useful since you looped over them before)